pkg/operator/configobservation/configobservercontroller/observe_config_controller.go

```diff
@@ -62,6 +64,7 @@ func NewConfigObserver(
 		images.ObserveAllowedRegistriesForImport,
 		ingresses.ObserveIngressDomain,
 		etcdobserver.ObserveStorageURLs,
+		etcdendpoints.ObserveStorageURLs,
```

Comment:
Please add test that combines the expected output of both just to be sure they don't step on each other.
I don't remember the exact rules `merge.Merge` (from https://github.com/openshift/cluster-kube-apiserver-operator/blob/22501ab373f961b9b5519803dc031830dd02defb/vendor/github.com/openshift/library-go/pkg/operator/configobserver/config_observer_controller.go#L92) uses.

Reply:
https://github.com/openshift/cluster-kube-apiserver-operator/pull/859 Over in the sibling PR I refactored to an approach that keeps the observer code separate but chooses the impl using a simple adapter function up front, take a look and comment over there and whatever approach we land on I'll port to this PR